src/zenml/models/v2/core/model.py

```diff
@@ -0,0 +1,329 @@
+
+    # Body and metadata properties
+    @property
+    def tags(self) -> List[TagResponseModel]:
+        """The `tags` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_body().tags
+
+    @property
+    def latest_version(self) -> Optional[str]:
+        """The `latest_version` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_body().latest_version
+
+    @property
+    def created(self) -> datetime:
+        """The `created` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_body().created
+
+    @property
+    def updated(self) -> datetime:
+        """The `updated` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_body().updated
+
+    @property
+    def license(self) -> Optional[str]:
+        """The `license` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().license
+
+    @property
+    def description(self) -> Optional[str]:
+        """The `description` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().description
+
+    @property
+    def audience(self) -> Optional[str]:
+        """The `audience` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().audience
+
+    @property
+    def use_cases(self) -> Optional[str]:
+        """The `use_cases` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().use_cases
+
+    @property
+    def limitations(self) -> Optional[str]:
+        """The `limitations` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().limitations
+
+    @property
+    def trade_offs(self) -> Optional[str]:
+        """The `trade_offs` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().trade_offs
+
+    @property
+    def ethics(self) -> Optional[str]:
+        """The `ethics` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().ethics
+
+    # Helper functions
+    @property
+    def versions(self) -> List["ModelVersion"]:
+        """List all versions of the model.
+
+        Returns:
+            The list of all model version.
+        """
+        from zenml.client import Client
+
+        client = Client()
+        model_versions = client.list_model_versions(
+            model_name_or_id=self.id, page=1
+        )
+        ret = [
+            mv.to_model_version(suppress_class_validation_warnings=True)
+            for mv in model_versions.items
+        ]
+        for i in range(2, model_versions.total_pages + 1):
```

Comment:
What is this supposed to do?

It's somehow looping over the amount of pages but using the model versions of the first page?

Reply:
What is the plan for this, I'm fine creating a separate PR but that just runs an additional set of tests. I think we should just clarify with @avishniakov what this is supposed to do and fix it. If this should just return a list of all model versions (not paginated), then we simply use `pagination_utils.depaginate(...)` and convert the results to `ModelVersion`, should be an easy fix no?

Reply:
Yes, this can be substituted with `pagination_utils.depaginate` 👍🏼